### ISSUE TITLE
CVE 2017-17805 test (af_alg02) is broken on kernels < 4.14

### ISSUE THREAD

**Issue by pevik** [linux-test-project/ltp#518]
CVE 2017-17805 test (af_alg02) expects `read()` to return, but is only in kernel with commit [2d97591ef43d](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=2d97591ef43d0587be22ad1b0d758d6df4999a0b) ("crypto: af_alg - consolidation of duplicate code"), added in v4.13-rc1. Therefore tests before fails after timeout as `read()` doesn't return:

```
[pid 12880] rt_sigreturn({mask=[]})     = 61
[pid 12880] wait4(12881,  <unfinished ...>
[pid 12881] brk(NULL)                   = 0x133d000
[pid 12881] brk(0x135e000)              = 0x135e000
[pid 12881] getpid()                    = 12881
[pid 12881] getpid()                    = 12881
[pid 12881] socket(AF_ALG, SOCK_SEQPACKET, 0) = 3
[pid 12881] bind(3, {sa_family=AF_ALG, sa_data="skcipher\0\0\0\0\0\0\0\0\0\0\0\0\0\0salsa20\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"}, 88) = 0
[pid 12881] setsockopt(3, SOL_ALG, 1, "g\306isQ\377J\354)\315\272\253\362\373\343F", 16) = 0
[pid 12881] accept(3, NULL, NULL)       = 4
[pid 12881] close(3)                    = 0
[pid 12881] read(4, 
```
@metan-ucw suggested to run the test in a child and handle the timeouting in parent.

**Comment by pevik** [comment]
@smuckle FYI

**Comment by pevik** [comment]
Fixed by c43a39a43 crypto/af_alg02: fixed read() being stuck, thanks Christian :)
